circe-akka-serializer/src/main/scala/org/virtuslab/ash/circe/CirceAkkaSerializer.scala

```diff
@@ -119,22 +124,26 @@ abstract class CirceAkkaSerializer[Ser <: AnyRef: ClassTag](system: ExtendedActo
     val foundSerializables = reflections.getSubTypesOf(classTag[Ser].runtimeClass).asScala.filterNot(_.isInterface)
     foundSerializables.foreach { clazz =>
       try {
-        val codec = codecsMap(parents(clazz.getName))
-        assert(codec._1 != null)
-        assert(codec._2 != null)
+        codecsMap(parents(clazz.getName))
       } catch {
         case e @ (_: NoSuchElementException | _: NotSerializableException) =>
           log.error(
             e,
             s"No codec found for [{}] class. Call Register[A] for this class or its supertype and append result to codecs.",
             clazz.getName)
-        case e: AssertionError =>
-          log.error(
-            e,
-            "Codec for [{}] is null. If this codec is custom defined, use def or lazy val instead of val.",
-            clazz.getName)
       }
     }
   }
+
+  private def checkCodecsForNull(): Unit = {
+    codecs.foreach { registration =>
+      val (tag, (encoder, decoder)) = registration
+      if (encoder == null || decoder == null)
+        throw new AssertionError(
+          s"Codec for [${tag.tpe.typeSymbol.fullName}] is null. If this codec is custom defined, use def or lazy val instead of val.")
```

Comment:
```suggestion
          s"Codec for [${tag.tpe.typeSymbol.fullName}] is null. If this codec is custom defined, declare it as a def or lazy val instead of val.")
```